src/storage/lock_manager.rs

```diff
@@ -10,9 +12,13 @@ pub struct Lock {
     pub hash: u64,
 }
 
+/// DiagnosticContext is for diagnosing problems about locks
 #[derive(Clone, Default)]
 pub struct DiagnosticContext {
+    /// the key we care about
     pub key: Vec<u8>,
+    /// this tag is used for keep related keys together
+    /// if the client is TiDB, this is usually the SQL statement which affect the key
```

Comment:
I think I didn't understand what this means. Now the resource_group_tag tags the kv requests so that we can track which statement (or any other related information) caused this request. It's an additional diagnostic information. And as it's said by Top SQL developers, this will also include the plan digest later.